app/components/syntax-highlighted-diff.ts

```diff
@@ -27,7 +43,19 @@ export default class SyntaxHighlightedDiffComponent extends Component<Signature>
   @tracked asyncHighlightedHTML: string | null = null;
   @tracked asyncHighlightedCode: string | null = null;
   @tracked lineNumberWithExpandedComments: number | null = null;
-  @tracked isDarkMode: boolean | undefined = undefined;
+
+  /**
+   * Current Dark Mode environment setting, reported by `is-dark-mode` modifier
+   * and used by `highlightCode` task to decide which mode to format code with
+   * @private
+   */
+  #isDarkMode: boolean | undefined = undefined;
+
+  /**
+   * The last Dark Mode environment setting that was used to format code
+   * @private
+   */
+  #isDarkModeRendered: boolean | undefined = undefined;
```

Comment:
In terms of purpose, this seems similar to `asyncHighlightedCode`. Maybe we can name this similar to that?

Reply:
Renamed this to `#lastHighlightCodeParams`, and it's now an instance of `HighlightCodeParamsCache` class, which implements `HighlightCodeParams` interface.